crates/sui-node/src/lib.rs

```diff
@@ -522,6 +522,8 @@ impl SuiNode {
         });
 
         let certified_checkpoint_output = SendCheckpointToStateSync::new(state_sync_handle);
+        // Note that this is constant and not a config as validators must have this set to the same value, otherwise they *will* fork
+        const MAX_TRANSACTIONS_PER_CHECKPOINT: usize = 1000;
```

Comment:
it would be great to have a way to override this for tests, so that we can exercise the max-txns code path

Reply:
maybe we can have it as a config param, but hard-code an override in sui-node/src/main.rs?

Reply:
We do excersize it in unit tests for checkpoint builder...

Reply:
but we can't exercise it in integration/e2e tests, right? unless we send tons of txns?

Reply:
Ok, I reduced it for e2e tests. With how code currently organized it is hard to just pass it as a parameter to SuiNode, but I used conditional compilation to reduce the value in e2e tests

Reply:
Why do we need to have this in integration/e2e tests? i.e. unit tests seem sufficient?

Reply:
we don't _need_ it, its a nice to have though. either way this is fine for now.